src/scripts/main.js

```diff
@@ -1,3 +1,302 @@
+      parseInt(employeeAge.value) < 18 ||
+      parseInt(employeeAge.value) > 90
+    ) {
+      pushNotification(
+        150,
+        10,
+        "Age input error",
+        "Age must not be less than 18 and more than 90",
+        "error"
+      );
+
+      return;
+    }
+
+    if (employeeSalary.value.trim() === "") {
+      pushNotification(
+        150,
+        10,
+        "Salary input error",
+        "Salary must not be empty",
+        "error"
+      );
+
+      return;
+    }
+
+    const newEmployeeRow = `
+      <tr>
+        <td>${employeeName.value}</td>
+        <td>${employeePosition.value}</td>
+        <td>${employeeOffice.value}</td>
+        <td>${employeeAge.value}</td>
+        <td>$${parseInt(employeeSalary.value).toLocaleString("en-US")}</td>
+      </tr>
+    `;
+
+    document
+      .querySelector("tbody")
+      .insertAdjacentHTML("beforeend", newEmployeeRow);
+
+    pushNotification(
+      150,
+      10,
+      "Success!",
+      "New employee successfully added to the table",
+      "success"
+    );
+
+    employeeName.value = "";
+    employeePosition.value = "";
+    employeeOffice.value = "Tokyo";
+    employeeAge.value = "";
+    employeeSalary.value = "";
+  });
+}
+
+function pushNotification(posTop, posRight, title, description, type) {
+  const body = document.querySelector("body");
+  const message = document.createElement("div");
+
+  message.classList = `notification ${type}`;
+  message.dataset.qa = "notification";
+  message.style.top = posTop + "px";
+  message.style.right = posRight + "px";
+
+  const messageTitle = document.createElement("h2");
+
+  messageTitle.className = "title";
+  messageTitle.textContent = title;
+
+  const messageDescription = document.createElement("p");
+
+  messageDescription.textContent = description;
+
+  message.append(messageTitle, messageDescription);
+  body.append(message);
+
+  setTimeout(() => {
+    message.style.visibility = "hidden";
+  }, 2000);
+}
+
+function editTableCell(table) {
+  const tbody = table.querySelector("tbody");
+  const input = document.createElement("input");
+
+  let isEditing = false;
+  let editedCell;
+  let previousTextContent = "";
+
+  tbody.addEventListener("dblclick", (e) => {
+    if (!isEditing && e.target.tagName === "TD") {
+      isEditing = true;
+      editedCell = e.target;
+      previousTextContent = editedCell.textContent;
+      input.classList = "cell-input";
+      input.value = previousTextContent;
+      editedCell.textContent = "";
+      editedCell.appendChild(input);
+      input.focus();
+    }
+  });
+
+  tbody.addEventListener("keydown", (e) => {
+    if (e.code !== "Enter" || !isEditing) {
+      return;
+    }
+
+    setEditedCellValue();
+  });
+
+  input.addEventListener("blur", (e) => {
+    setEditedCellValue();
+  });
+
+  function setEditedCellValue() {
+    if (isEditing) {
+      editedCell.textContent =
+        input.value.trim() !== "" ? input.value.trim() : previousTextContent;
```

Comment:
```suggestion
        input.value.trim() || previousTextContent;
```

Reply:
also, it would be nice to format the value (salary) after editing

![image](https://user-images.githubusercontent.com/86923089/215557352-e3b8c7dc-9b34-4d03-beb7-c5f21b4f047f.png)

Reply:
Done